setup.py

```diff
@@ -1,10 +1,108 @@
+import logging
+import os
+
 import setuptools
+from torch.utils import cpp_extension
+
+NAME = "bayes3d"
+VERSION = "0.0.1"
+HOME_DIR = os.path.dirname(__file__)
+CPP_SRC_DIR = os.path.join(HOME_DIR, "bayes3d", "rendering", "nvdiffrast", "common")
+
+with open(os.path.join(HOME_DIR, "requirements.txt")) as f:
+    REQUIRED_PACKAGES = f.read().splitlines()
+
+
+# Nvdiffrast compilation setup
```

Comment:
this looks great, but I wonder if there is a way to limit this work to only occur on installation. Are there setup.py tasks where we don't want all of this to get triggered? reading versions or something like that? I'm looking into this now...

Reply:
My understanding is that `setup.py` will only be executed on installation (i.e. when running `pip install .`). Users are not expected to run `setup.py` manually, and this script wouldn't be executed when we `import bayes3d` during run time either. (so the `if __name__ == "__main__"` condition is actually redundant in this file)